### PR TITLE
FDS-655 vertical forward links implemented

### PR DESCRIPTION
**Pull request by vikas-cldcvr**
### Checklist for raising a PR

- [ ] Gone through UX documnetation for adding new features.
- [ ] All necessary unit tests covered.
- [ ] Required comments added for generating component manifest file? you can find details [here](https://custom-elements-manifest.open-wc.org/analyzer/getting-started/)
- [ ] Did you check the contributing doc?
- [ ] Did you check the existing issues for similar queries?

### Describe your PR

# @ollion/flow-core

### Improvements

- `f-icon-button` box-sizing updated.

# @ollion/flow-table

### Bug Fixes

- `f-tcell` aligment in `f-table-schema` fixed.

# @ollion/flow-lineage

### Features

- `direction="vertical"` implemented in `f-lineage`.

Screenshot 2024-03-19 at 11 27 06 AM

### CODE REVIEW

File: packages/flow-lineage/src/components/f-lineage/create/create-node-elements.ts [hunk 1]
```diff
@@ -120,12 +120,14 @@ export default function createNodeElements(
...
 				 */
-				const maxYWhenScrollBar = nodeElement.y + nodeSize.height + maxChildrenHeight;
-				if (nodeElement.hasScrollbaleChildren && levelPointer.y > maxYWhenScrollBar) {
-					levelPointer.maxY = maxYWhenScrollBar;
-					nodeElement.childrenYMax = maxYWhenScrollBar;
-				} else if (levelPointer.y > (levelPointer.maxY ?? 0)) {
-					levelPointer.maxY = levelPointer.y;
+				if (!nodeElement.fHideChildren) {
+					const maxYWhenScrollBar = nodeElement.y + nodeSize.height + maxChildrenHeight;
+					if (nodeElement.hasScrollbaleChildren && levelPointer.y > maxYWhenScrollBar) {
```
Comment (NikhilVerma) [inline packages/flow-lineage/src/components/f-lineage/create/create-node-elements.ts:125]: `hasScrollableChildren`

File: packages/flow-lineage/src/components/f-lineage/draw/draw-nodes.ts [hunk 2]
```diff
@@ -116,10 +116,16 @@ export default function drawNodes(params: DrawLineageParams) {
 					let childHeight = d.childrenYMax - (d.y + nodeSize.height);
 
 					// finding all nodes below children
-					const nodesToUpdate = lineage.nodes.filter(
+					let nodesToUpdate = lineage.nodes.filter(
```
Comment (NikhilVerma) [inline packages/flow-lineage/src/components/f-lineage/draw/draw-nodes.ts:119]: Why do extra work here? If it's vertical we throw away the previous work

**Review by NikhilVerma — CHANGES_REQUESTED** [review]

File: packages/flow-lineage/src/components/f-lineage/draw/draw-nodes.ts (hunk 2 above)
Comment (vikas-cldcvr) [reply packages/flow-lineage/src/components/f-lineage/draw/draw-nodes.ts]: Code updated

File: packages/flow-lineage/src/components/f-lineage/create/create-node-elements.ts (hunk 1 above)
Comment (vikas-cldcvr) [reply packages/flow-lineage/src/components/f-lineage/create/create-node-elements.ts:125]: hasScrollableChildren is true if maxChildHeight is greater than the actual children's height and the second part is checking if the current y coordinate of a particular level is greater than the total height parent-child (This offset is calculated or used when children are collapsed and opened)

**Review by NikhilVerma — APPROVED** [review]

**Pull request merged**